staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -215,6 +227,8 @@ func NewBuilder(restClientGetter RESTClientGetter) *Builder {
 	return newBuilder(
 		restClientGetter.ToRESTConfig,
```

Comment:
Would it be more contained to 1) replicate the httpClient caching logic inside resource.Builder, 2) make it opt-in, and 3) only enable httpClient reuse from contexts we know are short-lived/safe (like kubectl `factoryImpl#NewBuilder()`)?

That would avoid leaking `stdinInUse` ugliness outside `resource.Builder`, ensure we don't regress `resource.Builder` behavior for callers other than kubectl, and make this more likely to be able to be backported